tests/common/fixtures/consistency_checker/query-asic.py

```diff
@@ -0,0 +1,484 @@
+        attr_value = attr.value.aclactiondatas16
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_IP_ADDRESS_LIST:
+        attr_value = attr.value.ipaddlist
+    else:
+        raise NotImplementedError(f"Unsupported attribute value type: {attr_metadata.attrvaluetype}")
+
+    if attr_metadata.isenum:
+        # Map to the string value if enum
+        enum_metadata = attr_metadata.enummetadata
+        attr_value = pysairedis.sai_metadata_get_enum_value_name(enum_metadata, attr_value)
+
+    return attr_value
+
+
+def query_asic_objects(query_objects) -> dict:
+    """
+    Query the ASIC for the attributes of the objects provided in deserialized JSON input file format.
+
+    :param query_objects: The deserialized JSON input file format
+    :return: The deserialized JSON output format
+    """
+
+    results = defaultdict(dict)
+
+    for oid_label_key, attributes in query_objects.items():
+        try:
+            logger.debug(f"Querying ASIC for object key {oid_label_key}")
+            oid = get_numeric_oid_from_label(oid_label_key)
+        except Exception as e:
+            err_msg = f"Failed to extract oid from label '{oid_label_key}': {e}"
+            logger.warning(err_msg)
+            for attribute in attributes:
+                results[oid_label_key][attribute] = {"success": False, "error": err_msg, "asicValue": None}
+            continue
+
+        for attribute in attributes:
+            try:
+                logger.debug(f"Querying ASIC object {oid_label_key} ({oid}) for attribute {attribute}")
+                attribute_oid = lookup_attribute_value_in_pysairedis(attribute)
+                asic_value = get_attribute_value_from_asic(oid, attribute_oid)
+
+                # Convert to str to match how values are represented in ASIC_DB
+                if asic_value in [True, False]:
+                    # ASIC_DB represents these as lowercase
+                    asic_value = str(asic_value).lower()
+                elif asic_value is None:
+                    asic_value = "NULL"
+                else:
+                    asic_value = str(asic_value)
+
+                # Success
+                results[oid_label_key][attribute] = {"asicValue": asic_value, "success": True}
+                logger.debug((f"Got ASIC object {oid_label_key} ({oid}) -> attribute {attribute} ({attribute_oid}) "
+                              f"value {asic_value}"))
+
+            except Exception as e:
+                err_msg = f"Failed to lookup attribute '{attribute}': {e}"
+                logger.warning(err_msg)
+                results[oid_label_key][attribute] = {"success": False, "error": err_msg, "asicValue": None}
+
+    return dict(results)
+
+
+def initialize_sai_api():
+    """
+    Initialize the SAI API
+    """
+    logger.info("Initializing SAI API")
+    profileMap = dict()
+    profileMap[pysairedis.SAI_REDIS_KEY_ENABLE_CLIENT] = "true"
+    status = pysairedis.sai_api_initialize(0, profileMap)
```

Comment:
Checking for pysairedis - is 0 for flags which should be zero as per sai spec?

https://github.com/opencomputeproject/SAI/blob/a0ae84d9366d35bfc50738c3b29e03387bf85c7c/inc/sai.h#L226